portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -46,6 +48,9 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     ArrayList<String> commentEntries = new ArrayList<String>();
     for (Entity entity : results.asIterable()) {
       String commentName = (String) entity.getProperty("name");
+      if(commentName.isEmpty()) {
```

Comment:
optional: You could alternatively fill the user's nickname for name from UserService if comment did not have a name. However, I see that this can be done only during post (when you obtain the userService). Though this does not guarantee to always have a name in the datastore.